x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/parser/ExpressionTests.java

```diff
@@ -188,6 +188,95 @@ public void testTripleDoubleQuotedUnescapedString() {
+                () -> expr("\"\\u001\""));
+        assertEquals("line 1:1: token recognition error at: '\"\\u001\"'", e.getMessage());
+
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("\"\\u{D}\""));
+        assertEquals("line 1:2: Unicode sequence in curly braces should use [2-8] hex digits, [\\u{D}] has [1]", e.getMessage());
+
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("\"\\u{00AFD1431}\""));
+        assertEquals("line 1:2: Unicode sequence in curly braces should use [2-8] hex digits, [\\u{00AFD1431}] has [9]", e.getMessage());
+
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("\"\\u{}\""));
+        assertEquals("line 1:1: token recognition error at: '\"\\u{}'", e.getMessage());
+
+        // Missing curly brace
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("\"\\u{DA12\""));
+        assertEquals("line 1:1: token recognition error at: '\"\\u{DA12\"'", e.getMessage());
+
+        // Missing opening curly brace but with 4 hex digits is accepted,
+        // as the closing brace on the 5th position is considered a normal char.
+        assertEquals(new Literal(null, "ǰ}", DataTypes.KEYWORD), expr("\"\\u01f0}\""));
+
+        // Null (all zeros escaped sequence)
+        StringBuilder zeros = new StringBuilder("0");
+        for (int i = 1; i < 8; i++) {
+            zeros.append("0");
+            e = expectThrows(ParsingException.class, "Expected syntax error",
+                    () -> expr("\"\\u{" + zeros.toString() + "}\""));
+            assertEquals("line 1:2: Unicode sequence results in null", e.getMessage());
+        }
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("\"\\u0000\""));
+        assertEquals("line 1:2: Unicode sequence results in null", e.getMessage());
+
+        // Inv
+        e = expectThrows(ParsingException.class, "Expected syntax error",
+                () -> expr("\"\\u{10000000}\""));
+        assertEquals("line 1:2: Invalid unicode character code [10000000]", e.getMessage());
+    }
+
+    public void testStringWithUnicodeEscapedChars() {
+        String strPadding = randomAlphaOfLength(randomInt(10));
+        String[][] strings = new String[][] {
+            { "\\u0021", "!" },
+            { "\\u{41}", "A" },
+            { "\\u{075}", "u" },
+            { "\\u{00eb}", "ë" },
+            { "\\u{1f0}", "ǰ" },
+            { "\\u0398", "Θ" },
+            { "\\u07e1", "ߡ" },
+            { "\\u{017e1}", "១" },
+            { "\\u2140", "⅀" },
+            { "\\u{02263}", "≣" },
+            { "\\u{0003289}", "㊉" },
+            { "\\u6d89", "涉" },
+            { "\\u{00007c71}", "籱" },
+            { "\\u{1680b}", "𖠋" },
+            { "\\u{01f4a9}", "💩" },
```

Comment:
(I guess this choice was asking for a note :slightly_smiling_face: )

Reply:
Just a random char 😇 ....